bin/functions

```diff
@@ -55,8 +57,12 @@ function retrieve-adoptopenjdk() {
           "https://api.adoptopenjdk.net/v3/assets/feature_releases/{13}/ga?architecture=${ARCHITECTURE}&os=${OS}&image_type=jdk&page=0&page_size=100&project=jdk&sort_order=ASC&vendor=adoptopenjdk"
           "https://api.adoptopenjdk.net/v3/assets/feature_releases/{14}/ga?architecture=${ARCHITECTURE}&os=${OS}&image_type=jdk&page=0&page_size=100&project=jdk&sort_order=ASC&vendor=adoptopenjdk")
 
-    # shellcheck disable=SC2046
-    if [[ -z "$(ls -A "${CACHE_DIR}"/adopt-*.json 2>/dev/null)" ]] || [[ $(set -- $(${STAT}) && echo "${1}") -le $(( $(date +%s) - 3600)) ]]
+    if ls "${CACHE_DIR}"/adopt-*.json > /dev/null 2>&1; then
+      mapfile -t cache_files <<<"$(ls "${CACHE_DIR}")"
+      mapfile -t cache_timestamps <<<"$(stat "${STAT_OPTS}" "${CACHE_DIR}"/*)"
+    fi
+
+    if [[ ! -v cache_files ]] || (( cache_timestamps[0] <= $(date +%s) - 3600 ))
```

Comment:
```suggestion
    local cache_timestamp
    local cache_files=("${CACHE_DIR}"/adopt-*.json)
    if [[ -f "${cache_files[0]}" ]]; then
      cache_timestamp="$(stat "${STAT_OPTS[@]}" "${cache_files[0]}")"
    else
      cache_timestamp=-1
    fi

    if (( cache_timestamp <= $(date +%s) - 3600 ))
```
In general, I think it might be better to not use `ls` as if any of the path has spaces, the code might work in unexpected ways. Initially, wanted to suggest to use a for loop with globbing but it seems like we just need 1 of the files to determine the cache timestamp, hence the change. The reason for the file existence check is because if the glob fails, the `adopt-*.json` would be put into the array. Thus, if the 1st file does not exists, it means that the glob is empty (no cache files). Alternatively, we could set the `nullglob` option, but we would have to unset it.

STAT_OPTS is changed to expand the array arguments into separate words, as per change above.

Reply:
By "glob fails", I mean if it did not manage to find any files.

Reply:
I also felt uncomfortable about using `ls` here, and had forgotten that a glob was a possible alternative. I'll definitely look into that and the `nullglob` option. Thank you.